parser/parser_test.go

```diff
@@ -526,9 +526,92 @@ func TestParse(t *testing.T) {
 						},
 					},
 					&ast.ExpressionStatement{
-						Expression: &ast.MemberExpression{
-							Object:   &ast.Identifier{Name: "m"},
-							Property: &ast.StringLiteral{Value: "key2"},
+						Expression: &ast.IndexExpression{
+							Object: &ast.Identifier{Name: "m"},
+							Index:  &ast.StringLiteral{Value: "key2"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "index expression",
+			raw:  `a[3]`,
+			want: &ast.Program{
+				Body: []ast.Statement{
+					&ast.ExpressionStatement{
+						Expression: &ast.IndexExpression{
+							Object: &ast.Identifier{Name: "a"},
+							Index:  &ast.IntegerLiteral{Value: 3},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "nested index expression",
+			raw:  `a[3][5]`,
+			want: &ast.Program{
+				Body: []ast.Statement{
+					&ast.ExpressionStatement{
+						Expression: &ast.IndexExpression{
+							Object: &ast.IndexExpression{
+								Object: &ast.Identifier{Name: "a"},
+								Index:  &ast.IntegerLiteral{Value: 3},
+							},
+							Index: &ast.IntegerLiteral{Value: 5},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "access indexed object returned from function call",
+			raw:  `f()[3]`,
+			want: &ast.Program{
+				Body: []ast.Statement{
+					&ast.ExpressionStatement{
+						Expression: &ast.IndexExpression{
+							Object: &ast.CallExpression{
+								Callee: &ast.Identifier{Name: "f"},
+							},
+							Index: &ast.IntegerLiteral{Value: 3},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "index with member expressions",
```

Comment:
This is the test case that would change based on the above. All of these are member expressions.